### DISCUSSION TITLE
set key to false can't disable the key

### DISCUSSION THREAD

**Discussion by FakeKuryr** [LazyVim/LazyVim#1040] [Q&A]
<img width="1028" alt="image" src="https://github.com/LazyVim/LazyVim/assets/32233191/93b34f92-ddd7-4e0b-a966-6e98b4dd8c54">
I write leap.lua according the  https://www.lazyvim.org/plugins#%EF%B8%8F-adding--disabling-plugin-keymaps
when I press S， it's already search pre
<img width="1347" alt="image" src="https://github.com/LazyVim/LazyVim/assets/32233191/8ae910e4-6245-4f43-bf04-3f126a6ccb31">

**Comment by folke** [comment]
You'll have to override the `config` instead since leap adds those mappings.

The keys in the spec are only there to provide lazy loading

**Reply by 2dogchan** [reply]
Similar question, I should override then `flash.nvim` config to disable the `s` key, right?
The following code works unexpectly.
```lua
  {
    "folke/flash.nvim",
    keys = { { "s", false } },
  },
```

**Reply by folke** [reply]
Yep, for flash (or any other plugin that defines keys), that's how you do it

**Reply by sagarPakhrin** [reply]
@folke the default behavior of 's' and 'S' is preety useful. seems like a lot of people wants to change the config to something different. 
Wouldn't it be better to have a different key instead of 's' and 'S' ?
At least it would be less confusing for beginners imo.

**Reply by dpetka2001** [reply]
You can just disable them in your own spec. Simple as that. Also, you can do the same with `cl` like with default `s` Neovim/Vim behavior. Maintainer is out for the entire month, as mentioned in the sticky issue, so don't take it personal since he won't be able to respond to you until he comes back.

**Reply by swelljoe** [reply]
I can't make the above work. I've added that to a new lua file in `config`, `plugins`, and in `config/keymaps.lua` and `s` continues to invoke flash.

**Reply by folke** [reply]
https://www.lazyvim.org/configuration/keymaps

```lua
 {
    "folke/flash.nvim",
    keys = { { "s", mode = { "n", "x", "o" }, false } },
  },
```

**Reply by swelljoe** [reply]
Thank you for the example. I'm unfortunately still getting flash invoked with `s` after adding this to `config/keymaps.lua` (with `return {}` around it. I'll keep experimenting.

**Reply by folke** [reply]
you need to  return that somewhere in your plugin specs. You really should read the docs to better understand how LazyVim / lazy.nvim works.

**Reply by swelljoe** [reply]
I've been reading this weekend and last, I'm just slow to understand it, I guess.

I moved it to my `plugins` lua file and that works. Thanks!

**Comment by Wizmann** [comment]
We should add this to the doc. I've searched for a hours to get this.

Besides, I still don't know how to "override the config" after reading this discussion.

---

Plus, thers's another people wants to remove the "s" binding but don't know how. ;)
https://www.reddit.com/r/neovim/comments/14h3pvk/keybindings_in_lazyvim/

**Reply by dpetka2001** [reply]
That's because people should understand that plugins that set keymaps with `lazy`'s `keys` handler, can be changed using the `keys` in their personal spec. But for plugins that specify keymaps in their internal configuration, then usually that should be changed in the `config` function after reading the plugin's corresponding README to understand how they can change keymaps. There's not only one way specifically to change keymaps, because plugins don't all inherit a specific configuration to set things.

**Reply by Wizmann** [reply]
Nice explain.

This is the best I can have for now to disable "s" binding. Please correct me if this is still doing wrong.

```lua
-- lua/plugins/flash.lua
return {
    {
        "folke/flash.nvim",
        keys = function()
            return {}
        end,
    },
}
```

**Reply by dpetka2001** [reply]
This way you disable all of the keys for `flash` plugin, not just `s`. If that's what you want, then yes. If you only want to disable `s`, then it has already been mentioned in another [comment](https://github.com/LazyVim/LazyVim/discussions/1040#discussioncomment-6351646) how to achieve this.
